clang/include/clang/Basic/TargetInfo.h

```diff
@@ -518,6 +518,16 @@ class TargetInfo : public TransferrableTargetInfo,
   /// getInt128Align() - Returns the alignment of Int128.
   unsigned getInt128Align() const { return Int128Align; }
 
+  /// getBitIntAlign/Width - Return aligned size of '_BitInt' and
+  /// 'unsigned _BitInt' for this target, in bits.
+  unsigned getBitIntWidth(unsigned NumBits) const {
+    return llvm::alignTo(NumBits, getBitIntAlign(NumBits));
+  }
+  virtual unsigned getBitIntAlign(unsigned NumBits) const {
```

Comment:
Instead of making this virtual, maybe it makes sense to add a field `std::optional<unsigned> BitIntMaxAlign` to TargetInfo?  I expect the logic here besides the max width to be consistent across all targets.

Reply:
I am not sure I understand how this idea would work. Could you please elaborate on what do you mean ?

Reply:
```
unsigned getBitIntMaxAlign() {
  return BitIntMaxAlign.or(LongLongAlign);
}
unsigned getBitIntAlign(unsigned NumBits) const {
    return std::clamp<unsigned>(llvm::PowerOf2Ceil(NumBits), getCharWidth(),
                                getBitIntMaxAlign());
}
```

Reply:
Done